src/components/Camera/index.js

```diff
@@ -65,23 +63,23 @@ export class CameraPure extends React.Component<CameraPureType> {
   }
 
   render() {
-    const {method, title, subTitle, onUploadFallback, onFallbackClick, hasError,
-      onUserMedia, onFailure, webcamRef, isFullScreen, i18n, video} = this.props;
+    const {method, title, subTitle, onUploadFallback, onFallbackClick, error, errorAction, className,
+      onUserMedia, onFailure, webcamRef, isFullScreen, isFullScreenDesktop, i18n, video} = this.props;
     return (
-      <div className={style.camera}>
-        <Title {...{title, subTitle, isFullScreen}} smaller={true}/>
+      <div className={classNames(style.camera, className)}>
+        <Title {...{title, subTitle, isFullScreen, isFullScreenDesktop}} smaller={true}/>
         <div className={classNames(style["video-overlay"], {[style.overlayFullScreen]: isFullScreen})}>
           {
-            hasError ?
+            error ?
```

Comment:
We are passing a prop down from the `Capture` parent component called `error`. This means that with this change we will now display any error passed down from the parent, even backend ones. This will interfere with the current autocapture behaviour where we keep sending frames to the document detect service until it does not return a validation error. `hasError` was only checking for camera errors, not for backend ones and I think we should preserve that behavior. Maybe you can rename it to `hasCameraError`.

Reply:
While completely agreeing with the need for renaming this, in my opinion is a symptom of a bigger problem, that is how props travelling far down the tree are exposed to intermediate components leaving the door open to:
- obscuring data to children by mistake (what I've done now)
- behave in certain ways depending on information not originally intended for their own consumption
- breaking encapsulation by implicit knowledge of responsibilities of other components 

Not intending any actions from the point I'm trying to make, just expressing a thought.

Reply:
I totally agree, we should not pass props that are not intended for a specific child component. We should avoid that pattern and extend the use of `flow` to every component.